OakCityV2/TableViewController.m

```diff
@@ -0,0 +1,216 @@
+    //Fetch JSON
+    NSString *urlAsString = [NSString stringWithFormat:@"https://jobs.github.com/positions.json?description=%@&location=%@", LANGUAGE, TOWN];
+    NSURL *url = [NSURL URLWithString:urlAsString];
+    NSURLRequest *request = [NSURLRequest requestWithURL: url];
+    AFHTTPRequestOperation *operation = [[AFHTTPRequestOperation alloc] initWithRequest:request];
+    operation.responseSerializer = [AFJSONResponseSerializer serializer];
+    
+    //Parse JSON
+    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject) {
+        self.jobs  = [NSMutableArray new];
+        NSArray *responseArray = (NSArray *)responseObject;
+        
+        for (NSDictionary *responseDict in responseArray) {
+            Job* job = [[Job alloc] initWithDictionary:responseDict];
+            [self.jobs addObject:job];
+        }
+        
+        if([self.jobs count] > 0) {
+            [self.tableView reloadData];
+        }
+        
+        //Alert on failure to fetch JSON
+        else {
+            UIAlertView* alert = [[UIAlertView alloc]
+                                  initWithTitle: @"Failed to retrieve data" message: nil delegate: self
+                                  cancelButtonTitle: @"cancel" otherButtonTitles: @"Retry", nil];
+            [alert show];
+        }
+    }
+     
+    //Upon failure
+    failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:@"Error"
+                                                             message:[error localizedDescription]
+                                                            delegate:nil
+                                                   cancelButtonTitle:@"Ok"
+                                                   otherButtonTitles:nil];
+        [alertView show];
+    }];
+    [operation start];
+    [sender endRefreshing];
+}
+
+- (NSInteger)numberOfSectionsInTableView:tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView
+ numberOfRowsInSection:(NSInteger)section {
+    return self.jobs.count;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView
+estimatedHeightForRowAtIndexPath:(NSIndexPath *)indexPath {
+    return UITableViewAutomaticDimension;
+}
+
+//Fill TableView
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    DetailCell *cell = [tableView dequeueReusableCellWithIdentifier:@"DetailCell" forIndexPath:indexPath];
+    
+    Job *aJob = self.jobs[indexPath.row];
+    [cell.titleLabel setText:aJob.title];
+    [cell.companyLabel setText:aJob.company];
+    if (aJob.logo != (NSString *)[NSNull null]) {
+        [cell.logoImageView setImageWithURL:[NSURL URLWithString:aJob.logo] placeholderImage:[UIImage imageNamed:@"placeholder.jpg"]];
+    }
+    else {
+        [cell.logoImageView setImage:[UIImage imageNamed:@"placeholder.jpg"]];
+    }
+    return cell;
```

Comment:
You should push off all of this detail work of setting labels and stuff to the DetailCell class.  In this method, you really just want to dequeue the cell, set cell.job = aJob and return the cell.  Let the cell handle taking the data object and assigning the various bits of data to it's UI.